src/DBConnection.cpp

```diff
@@ -371,22 +385,14 @@ sqlite3_stmt *DBConnection::Prepare(enum StatementID id, const char *sql)
    return stmt;
 }
 
-void DBConnection::CheckpointThread()
+void DBConnection::CheckpointThread(sqlite3 *db)
 {
-   // Open another connection to the DB to prevent blocking the main thread.
-   //
-   // If it fails, then we won't checkpoint until the main thread closes
-   // the associated DB.
-   sqlite3 *db = nullptr;
-   const auto name = sqlite3_db_filename(mDB, nullptr);
+   int rc = SQLITE_OK;
    bool giveUp = false;
-
-   auto rc = sqlite3_open(name, &db);
-   if (rc == SQLITE_OK)
+   const char *name = nullptr;
+   if (db)
```

Comment:
Really shouldn't matter since the thread won't actually be started if the open fails.  The "if (db)" at line 393 and the associated "else" isn't really needed since that's already handled in the main thread, though it should be logged as you suggested in your other comment.